app/src/main/java/co/ostorlab/insecure_app/bugs/calls/PackageContextCall.java

```diff
@@ -0,0 +1,56 @@
+package co.ostorlab.insecure_app.bugs.calls;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.util.Log;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+import co.ostorlab.insecure_app.BugRule;
+
+
+
+
+public final class PackageContextCall extends BugRule {
+
+    private static final String TAG = PackageContextCall.class.toString();
+
+
+    @Override
+    public String getDescription() {
+        return "Use of third-party packages";
+    }
+
+    @Override
+    public void run() {
+        Context context = getContext();
+        PackageManager packageManager = context.getPackageManager();
+
+        List<PackageInfo> installedPackages = packageManager.getInstalledPackages(PackageManager.GET_META_DATA);
+
+        for (PackageInfo info : installedPackages) {
+            String packageName = info.packageName;
+
+            if (packageName.startsWith("co.ostorlab.plugins.")) {
+
+                try {
+                    Context packageContext = context.createPackageContext(packageName,
+                            Context.CONTEXT_INCLUDE_CODE | Context.CONTEXT_IGNORE_SECURITY);
+
+                    Class<?> loaderClass = packageContext.getClassLoader().loadClass("co.ostorlab.plugins.camera.Loader");
+                    Method updateMethod = loaderClass.getMethod("Update", Context.class);
+                    updateMethod.invoke(null, packageContext);
```

Comment:
Shouldn't these be controllable to have a vulnerability?

Reply:
Not necessarily, the attacker can mimic the loaded package information and have the application invoke a method from a rogue package in its context, the attack still however requires the malicious package to be installed on the victim's device, hence the limited impact of this vulnerability.